### PR TITLE
Process data in the created models

### PR DESCRIPTION
**Pull request by JonahKayizzi**
This pull request consists of the following;

- Created model classes for all entities
- Set up associations between models.
- Tested CRUD methods in IRB
- Created the following custom methods
`A method that returns the 3 most recent posts for a given user.`
`A method that updates the posts counter for a user.`
`A method which returns the 5 most recent comments for a given post.`
`A method that updates the comments counter for a post.`
`A method that updates the likes counter for a post.`

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class User < ApplicationRecord
+  has_many :posts
+  has_many :likes
+  has_many :comments
```
Comment (Ismailco) [inline app/models/user.rb]: - [ ] Kindly make sure to add `foreign keys` for the `posts`, `likes`, and `comments`
e.g
```ruby
has_many :posts, foreign_key: 'author_id'
has_many :comments, foreign_key: 'author_id'
has_many :likes, foreign_key: 'author_id'
```

File: app/models/comment.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :user, class_name: 'User', foreign_key: 'author_id'
+  belongs_to :post, class_name: 'Post', foreign_key: 'post_id'
+
+  def five_most_recent_comments_for_user(user)
+    user.comments.order(created_at: :desc).limit(5)
+  end
```
Comment (Ismailco) [inline app/models/comment.rb]: - [ ] This method for returning the 5 recent comments should be implemented in the `post` model, not in the comment model kindly make sure to fix this

File: app/models/post.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class Post < ApplicationRecord
+  belongs_to :user, class_name: 'User', foreign_key: 'author_id'
+  has_many :likes
+  has_many :comments
+
```
Comment (Ismailco) [inline app/models/post.rb:5]: - [ ] The post model should have a method that returns the 5 recent comments per post, kindly consider implementing this so you can return limited posts in your app

File: app/models/comment.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :user, class_name: 'User', foreign_key: 'author_id'
+  belongs_to :post, class_name: 'Post', foreign_key: 'post_id'
+
+  def five_most_recent_comments_for_user(user)
```
Comment (Ismailco) [inline app/models/comment.rb]: - [ ] The comment model should include a method that updates the comments counter for a post, kindly consider implementing this feature

File: app/models/like.rb [hunk 5]
```diff
@@ -0,0 +1,4 @@
+class Like < ApplicationRecord
```
Comment (Ismailco) [inline app/models/like.rb:1]: - [ ] The like model should include a method that updates the like counter for a post, kindly consider implementing this feature

File: app/models/post.rb [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+  def update_likes_counter_for_post
+    update(LikesCounter: likes.count)
+  end
```
Comment (Ismailco) [inline app/models/post.rb]: - [ ] This method for updating the like counter for a post should be implemented in the Like model kindly fix this

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @JonahKayizzi, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 6 above)
Comment (JonahKayizzi) [reply app/models/post.rb]: Updated

File: app/models/like.rb (hunk 5 above)
Comment (JonahKayizzi) [reply app/models/like.rb:1]: Updated

File: app/models/comment.rb (hunk 4 above)
Comment (JonahKayizzi) [reply app/models/comment.rb]: Updated

File: app/models/post.rb (hunk 3 above)
Comment (JonahKayizzi) [reply app/models/post.rb:5]: Updated

File: app/models/comment.rb (hunk 2 above)
Comment (JonahKayizzi) [reply app/models/comment.rb]: Updated

File: app/models/user.rb (hunk 1 above)
Comment (JonahKayizzi) [reply app/models/user.rb]: Updated

**Comment by JonahKayizzi** [comment]
Thanks for the suggestions, all have been updated

**Review by codecaiine — APPROVED** [review]
Hi @JonahKayizzi , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Great project implementation :heavy_check_mark: 
- No rubocop errors :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**